### PR TITLE
TST: Fix test_parquet failures for pyarrow 1.0

### PR DESCRIPTION
**Pull request by alimcmaster1**
- [x] closes https://github.com/pandas-dev/pandas/issues/35791
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

cc @martindurant -> looks like this is a pyarrow 1.0.0 compat issue (read_table uses the new API) - https://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html

I noticed the partition cols are casted from int64 -> int32 is that expected pyarrow behaviour? From the write_table docs looking at version 1.0/2.0 it suggests it is https://arrow.apache.org/docs/python/generated/pyarrow.parquet.write_table.html#pyarrow.parquet.write_table

### CODE REVIEW

**Comment by martindurant** [comment]
May be best asking @jorisvandenbossche about intended behaviour

**Comment by jorisvandenbossche** [comment]
@alimcmaster1 thanks for looking into this! 

Yes, so with pyarrow 1.0 and using the new datasets implementation, we kept the default for category type if your partition field is string, but not for integers (and indeed also the default now is int32, not int64)

So I suppose out other roundtrip tests are using string partition fields (since this test is the only one that is failing)

**Comment by jorisvandenbossche** [comment]
> So I suppose out other roundtrip tests are using string partition fields (since this test is the only one that is failing)

Actually it doesn't see we have other full roundtrip tests that use partitioning .. (only tests checking that it is written correctly)

File: pandas/tests/io/test_parquet.py [hunk 1]
```diff
@@ -557,13 +557,22 @@ def test_s3_roundtrip(self, df_compat, s3_resource, pa):
     @pytest.mark.parametrize("partition_col", [["A"], []])
     def test_s3_roundtrip_for_dir(self, df_compat, s3_resource, pa, partition_col):
         # GH #26388
-        # https://github.com/apache/arrow/blob/master/python/pyarrow/tests/test_parquet.py#L2716
-        # As per pyarrow partitioned columns become 'categorical' dtypes
+        expected_df = df_compat.copy()
+
+        # read_table uses the new Arrow Datasets API since pyarrow 1.0.0
+        # Previous behaviour was pyarrow partitioned columns become 'categorical' dtypes
```
Comment (jorisvandenbossche) [inline pandas/tests/io/test_parquet.py:563]: so only for integer columns the behaviour changed, for string columns it is still using category type

**Comment by alimcmaster1** [comment]
> @alimcmaster1 thanks for looking into this!
> 
> Yes, so with pyarrow 1.0 and using the new datasets implementation, we kept the default for category type if your partition field is string, but not for integers (and indeed also the default now is int32, not int64)
> 
> So I suppose out other roundtrip tests are using string partition fields (since this test is the only one that is failing)

Gotcha makes sense - thanks for the info!

**Comment by jreback** [comment]
@jorisvandenbossche lgtm.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@MeeseeksDev backport to 1.1.x

**Pull request merged**